runtime/nvqir/cutensornet/tensornet_state.cpp

```diff
@@ -321,10 +321,29 @@ TensorNetState::factorizeMPS(int64_t maxExtent, double absCutoff,
     throw std::runtime_error("ERROR: Insufficient workspace size on Device!");
   }
 
+  // Check whether we need host memory workspace
+  int64_t hostWorkspaceSize{0};
+  HANDLE_CUTN_ERROR(cutensornetWorkspaceGetMemorySize(
+      m_cutnHandle, workDesc, CUTENSORNET_WORKSIZE_PREF_RECOMMENDED,
+      CUTENSORNET_MEMSPACE_HOST, CUTENSORNET_WORKSPACE_SCRATCH,
+      &hostWorkspaceSize));
+
+  void *hostWork = nullptr;
+  if (hostWorkspaceSize > 0) {
+    hostWork = malloc(hostWorkspaceSize);
```

Comment:
It would probably be a good idea to check for a nullptr and throw an error message / exception in that case (or perform error mitigation if it's easy).

Reply:
It's a good point. Added a null check as bellow (it's best to just abort here since we don't really have an alternative).